test/cpp/interop/stress_interop_client.h

```diff
@@ -84,20 +85,26 @@ class WeightedRandomTestSelector {
 class StressTestInteropClient {
  public:
   StressTestInteropClient(int test_id, const grpc::string& server_address,
+                          std::shared_ptr<Channel> channel,
                           const WeightedRandomTestSelector& test_selector,
-                          long test_duration_secs, long sleep_duration_ms);
+                          long test_duration_secs, long sleep_duration_ms,
+                          long metrics_collection_interval_secs);
 
-  void MainLoop();  // The main function. Use this as the thread entry point.
+  // The main funciton. Use this as the thread entry point.
+  // qps_gauge is the Gauge to record the requests per second metric
+  void MainLoop(std::shared_ptr<Gauge> qps_gauge);
```

Comment:
Thanks for the link. After reading through that, I still feel using shared_ptr is probably better:

This is not a simple use case as mentioned in the article above (where caller and the callee functions are both executing on the same thread). In this case, we are calling function on a new thread. 

Moreover, unlike the cases mentioned in the above article (where the callee doesn't strictly need to share the ownership of the object), in our use case,  we really want to "share ownership" of the Gauge object between the metrics server (which maintains it in the gauges_ map in MetricsServiceImpl object and serves it up via RPC calls) and the stress test client which uses it to update the Gauge at various places (although in this case, it only updates it in one function)